src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/SubmissionsController.kt

```diff
@@ -79,6 +82,22 @@ class SubmissionsController(
     return ResponseEntity(HttpStatus.OK)
   }
 
+  override fun submissionsApplicationIdStatusUpdatesPost(
+    applicationId: UUID,
+    statusUpdate: Cas2ApplicationStatusUpdate,
+  ): ResponseEntity<Unit> {
+    if (!statusUpdateService.isValidStatus(statusUpdate)) {
+      return ResponseEntity(HttpStatus.NOT_FOUND)
```

Comment:
Are we doing the check twice if it's a valid status? here in the controller and then again in the service? Is that so we can return a 404 here?